packages/tasit-action/src/NFT.test.js

```diff
@@ -189,5 +205,103 @@ describe("TasitAction.NFT", () => {
       const balanceAfter = await fullNFT.balanceOf(bob.address);
       expect(balanceAfter.toNumber()).to.equal(1);
     });
+
+    it("should transfer an owned token using safeTransferFrom", async () => {
+      let senderBalance, receiverBalance;
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(ana.address);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(bob.address);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      txSubscription = fullNFT.safeTransferFrom(
+        ana.address,
+        bob.address,
+        tokenId
+      );
+
+      await txSubscription.waitForNonceToUpdate();
+
+      senderBalance = await fullNFT.balanceOf(ana.address);
+      expect(senderBalance.toNumber()).to.equal(0);
+
+      receiverBalance = await fullNFT.balanceOf(bob.address);
+      expect(receiverBalance.toNumber()).to.equal(1);
+    });
+
+    it("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Contract Subscription", async () => {
+      let senderBalance, receiverBalance;
+      const sender = ana.address;
+      const receiver = simpleStorageWithRemovedAddress;
+      const contractErrorFakeFn = sinon.fake();
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+
+      const contractErrorListener = message => {
+        const { error, eventName } = message;
+        expect(error.message).to.equal(
+          "Action with error: VM Exception while processing transaction: revert"
+        );
+        contractErrorFakeFn();
+      };
+
+      fullNFT.on("error", contractErrorListener);
+
+      txSubscription = fullNFT.safeTransferFrom(sender, receiver, tokenId);
+      await txSubscription.waitForNonceToUpdate();
+
+      expect(contractErrorFakeFn.called).to.be.true;
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
+
+      receiverBalance = await fullNFT.balanceOf(receiver);
+      expect(receiverBalance.toNumber()).to.equal(0);
+    });
+
+    it("shouldn't do a safeTransferFrom to a contract that doesn't implement `onERC721Received` - Action Subscription", async () => {
+      let senderBalance, receiverBalance;
+      const sender = ana.address;
+      const receiver = simpleStorageWithRemovedAddress;
+      const actionErrorFakeFn = sinon.fake();
+
+      fullNFT = new NFT(fullNFTAddress, ana);
+
+      senderBalance = await fullNFT.balanceOf(sender);
+      expect(senderBalance.toNumber()).to.equal(1);
```

Comment:
We could consider making a helper function maybe that does assertion before for a given sender and receiver in this test file that DRY's these tests up a bit

Reply:
TODO

Reply:
Done